infrastructure/restapi/src/main/java/tech/pegasys/teku/infrastructure/restapi/endpoints/EndpointMetadata.java

```diff
@@ -241,10 +241,10 @@ private String selectContentType(
     final String selectedType =
         ContentTypes.getResponseContentType(supportedTypes, acceptHeader)
             .orElse(defaultResponseContentType);
-    checkState(
-        supportedTypes.contains(selectedType),
-        "Response content type %s was selected but is not supported.",
-        selectedType);
+    if (!supportedTypes.contains(selectedType)) {
+      throw new IllegalArgumentException(
+          "Response content type " + selectedType + " was selected but is not supported.");
+    }
```

Comment:
@ajsutton Updated this with @rolfyone so that a bad request is given instead of an internal server error.

Reply:
Hmm, I think this should be an internal server error unless I've misunderstood the context.  I think this occurs when we've selected a response type to return that we don't actually support (most commonly because the default response type isn't a supported type).  That's entirely a code error and not a bad request. If the user specifies an unsupported type in `Accept` header we should just fall back to the default.

We'd only give a bad request if we didn't support the _request_ content type as then we can't parse the request content.

Reply:
we have an ssz request that only allows ssz, so requesting json we'd get a 500 back...

Reply:
Then we should set the default to be application/octet-stream so we send SSZ regardless of what's requested.

Reply:
We may not have exposed the concept of default content type much but I'm pretty sure it exists...

Reply:
`EndpointMetadataBuilder.defaultResponseType(ContentTypes.OCTET_STREAM)` should do it.

Reply:
potentially we need to override the default type so that in this case it can default to ssz if none is specified.. that'd be another option...

so @courtneyeh we can override default by 
```
.defaultResponseType("application/octet-stream")
```
but will still error 500 if the user requests application/json specifically

Reply:
Okay so its fine to get 500 if user requests application/json. I will add in that change.

Reply:
If the user requests application/json and we don't support it we should return ssz (we shouldn't declare json as a supported type at all which would make that just work I think)

Reply:
Same as for most endpoints if the user specifically sends `Accept: made-up/crap` we will respond with JSON.